python/sparkdl/graph/utils.py

```diff
@@ -0,0 +1,255 @@
+
+def get_tensor(graph, tfobj_or_name):
+    """
+    Get a tf.Tensor object
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
+    """
+    graph = validated_graph(graph)
+    if isinstance(tfobj_or_name, tf.Tensor):
+        return tfobj_or_name
+    name = tfobj_or_name
+    if isinstance(tfobj_or_name, tf.Operation):
+        name = tfobj_or_name.name
+    if not isinstance(name, six.string_types):
+        raise TypeError('invalid tensor request for {} of {}'.format(name, type(name)))
+    _tensor_name = as_tensor_name(name)
+    tnsr = graph.get_tensor_by_name(_tensor_name)
+    assert tnsr is not None, \
+        'cannot locate tensor {} in current graph'.format(_tensor_name)
+    return tnsr
+
+def as_tensor_name(name):
+    """
+    Derive tf.Tensor name from an op/tensor name
+
+    :param name: op name or tensor name
+    """
+    assert isinstance(name, six.string_types)
+    name_parts = name.split(":")
+    assert len(name_parts) <= 2
+    if len(name_parts) < 2:
+        name += ":0"
+    return name
+
+def as_op_name(name):
+    """
+    Derive tf.Operation name from an op/tensor name
+
+    :param name: op name or tensor name
+    """
+    assert isinstance(name, six.string_types)
+    name_parts = name.split(":")
+    assert len(name_parts) <= 2
+    return name_parts[0]
+
+def op_name(graph, tfobj_or_name):
+    """
+    Get the name of a tf.Operation
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
+    """
+    graph = validated_graph(graph)
+    return get_op(graph, tfobj_or_name).name
+
+def tensor_name(graph, tfobj_or_name):
+    """
+    Get the name of a tf.Tensor
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
+    """
+    graph = validated_graph(graph)
+    return get_tensor(graph, tfobj_or_name).name
+
+def validated_output(graph, tfobj_or_name):
+    """
+    Validate and return the output names useable GraphFunction
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
+    """
+    graph = validated_graph(graph)
+    return op_name(graph, tfobj_or_name)
+
+def validated_input(graph, tfobj_or_name):
+    """
+    Validate and return the input names useable GraphFunction
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
+    """
+    graph = validated_graph(graph)
+    name = op_name(graph, tfobj_or_name)
+    op = graph.get_operation_by_name(name)
+    assert 'Placeholder' == op.type, \
+        ('input must be Placeholder, but get', op.type)
+    return name
+
+def strip_and_freeze_until(fetches, graph, sess=None, return_graph=False):
+    """
+    Converting all variables into constants
```

Comment:
you should mention that it prunes out all the parts of the graph that are unreachable from the fetches.